### ISSUE TITLE
Palettes above 15 not setting correctly

### ISSUE THREAD

**Issue by gamax92** [MightyPirates/OpenComputers#292]
![Failure](http://i.imgur.com/lOtzvY4.png)

As shown in the above image, the palette didn't set in the second instance, and gave the wrong color.

This is both a Tier 3 screen and graphics card, and doubled check to make sure.

Also, a separate issue with the palettes:
http://i.imgur.com/lQbM2yT.png
Setting a palette index to a specific color, and then setting the foreground and background to said color, should result in said color being used. However OC picks a different color.

**Comment by fnuecke** [comment]
Fixed, thanks! For clarification, the actual bug here was that it didn't raise an error, but returned `nil, message` in the `setPaletteColor` until now, and corrupted the state a little in the `setForeground`/`setBackground` if the palette index was out of bounds. I made it raise an error to avoid this confusion in the future.

Note that tier 2 screens have 16 colors, so they have a 16 color palette, tier 3 screens have a mixed mode: 240 'fixed' colors (picked by calling `setForeground`/`setBackground` as before, using the hex code) and a 16 color palette, which is initialized to shades of gray.
